pkg/thing/gateway.go

```diff
@@ -48,48 +49,64 @@ var ErrCOAPServerAlreadyStarted = errors.New("CoAP server has already been start
 
 // ThingGateway represents the Thing Gateway
 type ThingGateway struct {
-	Thing     defaultThing
-	authCache *tokencache.Cache
+	gatewayThing     Thing
+	authCache        *tokencache.Cache
+	callbackHandlers []callback.Handler
 	// coap server
 	coapServer *coap.Server
 	coapChan   chan error
 	address    net.Addr
+	// AM connection
+	amConnection client.Connection
```

Comment:
I keep flip-flopping on this but I like having two separate connections avoids the Gateway's credentials being used for a Thing. When we review the Gateway creation API, we should allow the connection data to be different. I can imagine a Gateway using a different tree to a Thing for example.

Would `thingConnection` be a clearer name, it wasn't immediately obvious to me that this was for Things only.

Reply:
Did we not decide that things must always use the same tree as the Gateway? That is how the Thing API is/will be documented. We should only store common config with the connection. If config is specific to the thing then it should be stored with the thing.

The `amConnection` can be used for things and the gateway, because it does not hold credentials or handlers. The reason I added the `gatewayThing` is for convenience of registering the gateway. It should really be using the same connection. Maybe adding a `WithConnection` method to the thing builder?